presto-main/src/main/java/io/prestosql/sql/planner/ExpressionInterpreter.java

```diff
@@ -588,14 +588,12 @@ protected Object visitInPredicate(InPredicate node, Object context)
             boolean found = false;
             List<Object> values = new ArrayList<>(valueList.getValues().size());
             List<Type> types = new ArrayList<>(valueList.getValues().size());
-            List<Expression> originalValues = new ArrayList<>(valueList.getValues().size());
```

Comment:
> Remove special handling for types without a literal form

what about plugn-provided types? do they have a literal form?

Reply:
We decided not to worry about this for now, since it only affects these special types used for functions. Normal types provided by plugins work. If this is a problem for anyone, we can make it extensible later.